x86_symbolic/src/Data/Macaw/X86/Symbolic/ABI/SysV.hs

```diff
@@ -0,0 +1,219 @@
+{-|
+Copyright        : (c) Galois, Inc 2024
+Maintainer       : Langston Barrett <langston@galois.com>
+
+On x86_64 with the SysV ABI, the stack grows \"downwards\" from high addresses
+to low. The end of the stack is initialized with the ELF auxiliary vector (not
+modelled here), and functions expect the following data to be available above
+their stack frame (i.e., just above the address in @rsp@), from high addresses
+to low:
+
+* Their stack-spilled arguments
+* The return address
+
+The functions in this module support manipulation of a stack under these
+constraints. ABI-compatible machine code translated by Macaw manages the stack
+for itself, so this module is primarily helpful for initial setup of the stack,
+before starting symbolic execution.
+
+Helpful links:
+
+* <https://eli.thegreenplace.net/2011/09/06/stack-frame-layout-on-x86-64>
+* <https://wiki.osdev.org/System_V_ABI#x86-64>
+
+TODO: The stack is supposed to be 16-byte aligned before a @call@.
```

Comment:
Does this TODO still need to be addressed?

Reply:
Yes, but I might prefer to do so in a follow-up if that's alright. I'm not exactly sure what this even looks like (e.g., do you always put the return address 8 bytes before a 16-byte-aligned address?), so I'd have to do a bit more digging to figure it out. In any case, it doesn't seem to affect the test suite.

Reply:
https://github.com/GaloisInc/macaw/issues/436